### PR TITLE
zsh-fast-syntax-highlighting:init at 1.54

### PR DESCRIPTION
**Pull request by pasqui23**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    cd "../themes"
+    install -dm0755 "$_themesdir"
+    install -m0644 -- *.ini "$_themesdir"
+  '';
+
+  meta = with lib; {
+    description = "Syntax-highlighting for Zshell";
+    homepage = https://github.com/zdharma/fast-syntax-highlighting ;
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ qyliss ];
```
Comment (alyssais) [inline pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix]:
```suggestion
```

I would not be a good maintainer of this software, considering I’ve never heard of it!

**Review by alyssais — DISMISSED** [review]

**Comment by timokau** [comment]
Thank you for your contribution, but please keep "PRs already reviewed" to only those PRs that have actually been reviewed ;)

File: pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
```
Comment (veprbl) [inline pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix]:
```suggestion
stdenvNoCC.mkDerivation {
```

File: pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    install -m0644 -- *.ch "$_chromadir"
+
+    cd "../themes"
+    install -dm0755 "$_themesdir"
+    install -m0644 -- *.ini "$_themesdir"
+  '';
+
+  meta = with lib; {
+    description = "Syntax-highlighting for Zshell";
+    homepage = https://github.com/zdharma/fast-syntax-highlighting ;
```
Comment (veprbl) [inline pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix]:
```suggestion
    homepage = "https://github.com/zdharma/fast-syntax-highlighting";
```
https://github.com/NixOS/rfcs/pull/45

File: pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "zsh-fast-syntax-highlighting";
+  version = "2019-07-20";
```
Comment (veprbl) [inline pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix]: According to https://nixos.org/nixpkgs/manual/#sec-package-naming this should read something like "unstable-2019-07-20"
Why use a specific revision if there are (newer) tagged releases? https://github.com/zdharma/fast-syntax-highlighting/releases

File: pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    sha256 = "1h65ydb90wzl0f2w0q3sgc77l1zdn6w33fv8fg5nvx2ycl2xr43x";
+  };
+
+  #adapted from https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zsh-fast-syntax-highlighting-git
+  installPhase = ''
+    _plugindir="$out/share/zsh/site-functions"
+    _chromadir="$_plugindir/chroma"
+    _themesdir="$_plugindir/themes"
+
+    cd "${src}"
```
Comment (veprbl) [inline pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix]: Please don't use string interpolation, use the environment variable instead to allow users to override it.
```suggestion
    cd "$src"
```
Also this installPhase is a bit too complicated for it's functionality (copy files from `"$src"` to `"$out"/share/zsh/site-functions`)...

**Review by jonringer — COMMENTED** [review]
please squash all commits:

```
git reset HEAD~6
git add pkgs/
git commit --amend --no-edit
git push --force <fork> <branch>
```

**Comment by veprbl** [comment]
@GrahamcOfBorg build zsh-fast-syntax-highlighting

**Comment by veprbl** [comment]
@pasqui23 I took liberty to further simplify the expression. The "cd $src" line is actually not needed after `unpackPhase` creates a copy of the source code and harmful in cases when user tries to add `patches` or override `src` to point to a tarball. Also "install -m644" doesn't do anything for us, the write bit will be reset by Nix for files in the store. An "install -m444" would be fine, but we are already doing "cp" on some directories, so we might as well stick with it.

**Review by veprbl — APPROVED** [review]
Tested on NixOS, highlighting works.

**Comment by lucc** [comment]
@veprbl sorry to bother you in this old issue but how did you use the package? Did you just do 
```nix
{
  programs.zsh.interactiveShellInit = ''
    source ${pkgs.zsh-fast-syntax-highlighting}/share/zsh/site-functions/fast-syntax-highlighting.plugin.zsh
  '';
}
```
in your configuration.nix or is there a better way?

**Comment by veprbl** [comment]
@lucc, I never set this up on my system, just tested that it works to "source" it in zsh.

**Pull request merged**